src/ray/raylet/scheduling/cluster_resource_data.h

```diff
@@ -159,6 +162,15 @@ class NodeResources {
   /// Map containing custom resources. The key of each entry represents the
   /// custom resource ID.
   absl::flat_hash_map<int64_t, ResourceCapacity> custom_resources;
+  /// Amongst CPU, memory, and object store memory, calculate the utilization percentage
+  /// of each resource and return the highest.
+  float CalculateCriticalResourceUtilization() const;
+  /// Returns true if the node has the available resources to run the task.
+  /// Note: This doesn't account for the binpacking of unit resources.
+  bool IsAvailable(const TaskRequest &task_req) const;
```

Comment:
Do we still need `IsFeasible` in the `cluster_resource_scheduler` after this? (except for the legacy logic)

Reply:
I think it's only used by the legacy scheduler.

Reply:
Maybe just adding TODO comments on those functions so that we can easily clean up later?